packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionBarcodeDetectorModule.m

```diff
@@ -0,0 +1,239 @@
+//
+/**
+ * Copyright (c) 2016-present Invertase Limited & Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this library except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#import <React/RCTUtils.h>
+#import "RNFBSharedUtils.h"
+#import "RNFBMLVisionBarcodeDetectorModule.h"
+#import "RNFBMLVisionCommon.h"
+
+@implementation RNFBMLVisionBarcodeDetectorModule
+#pragma mark -
+#pragma mark Module Setup
+
+RCT_EXPORT_MODULE();
+
+#pragma mark -
+#pragma mark Firebase ML Kit Vision Methods
+
+RCT_EXPORT_METHOD(barcodeDetectorProcessImage:
+  (FIRApp *) firebaseApp
+    : (NSString *)filePath
+    : (NSDictionary *)barcodeDetectorOptions
+    : (RCTPromiseResolveBlock)resolve
+    : (RCTPromiseRejectBlock)reject
+) {
+  [RNFBMLVisionCommon UIImageForFilePath:filePath completion:^(NSArray *errorCodeMessageArray, UIImage *image) {
+    if (errorCodeMessageArray != nil) {
+      [RNFBSharedUtils rejectPromiseWithUserInfo:reject userInfo:(NSMutableDictionary *) @{
+          @"code": errorCodeMessageArray[0],
+          @"message": errorCodeMessageArray[1],
+      }];
+      return;
+    }
+
+    FIRVisionImage *visionImage = [[FIRVisionImage alloc] initWithImage:image];
+    FIRVision *vision = [FIRVision vision];
+
+    FIRVisionBarcodeDetectorOptions *options = [[FIRVisionBarcodeDetectorOptions alloc] initWithFormats: FIRVisionBarcodeFormatQRCode | FIRVisionBarcodeFormatAztec];
```

Comment:
How do we convert a list of ints (from JS) into this type? Left as per the example for now.